pkg/services/extsvcauth/registry/service.go

```diff
@@ -110,8 +112,12 @@ func (r *Registry) RemoveExternalService(ctx context.Context, name string) error
 
 	switch provider {
 	case extsvcauth.ServiceAccounts:
-		if !r.features.IsEnabled(ctx, featuremgmt.FlagExternalServiceAccounts) {
-			r.logger.Debug("Skipping External Service removal, flag disabled", "service", name, "flag", featuremgmt.FlagExternalServiceAccounts)
+		if !r.enabled {
+			r.logger.Warn("Skipping External Service authentication, flag disabled",
```

Comment:
nit: the warning is not fully accurate anymore - it's not just the flag that we're checking now. Maybe extend it to something like:
```suggestion
			r.logger.Warn("Skipping External Service authentication, flag or configuration option is disabled",
```